.github/workflows/ci.yml

```diff
@@ -84,23 +84,11 @@ jobs:
 
     strategy:
       matrix:
-        os: [windows-latest, macos-latest, ubuntu-latest]
-        include:
-          - os: windows-latest
-            shell: msys2 {0}
-            # GNU Windows is used as we need
-            # `unistd.h` and more in `cryptonight/`.
-            rust: stable-x86_64-pc-windows-gnu
-          - os: macos-latest
-            shell: bash
-            rust: stable
-          - os: ubuntu-latest
-            shell: bash
-            rust: stable
-
-    defaults:
-      run:
-        shell: ${{ matrix.shell }}
+        os: [
+            windows-2022, # EOL = 2026-10-13 | <https://endoflife.date/windows-server>
+            macos-15,     # EOL = 2027-09-16 | <https://endoflife.date/macos>
+            ubuntu-22.04  # EOL = 2027-04-01 | <https://endoflife.date/ubuntu>
```

Comment:
Since it's backwards compatible I'd prefer `windows-2025` but this is fine for now.

Ubuntu 22.04 has [glibc 2.35](https://launchpad.net/ubuntu/+source/glibc/2.35-0ubuntu3) which should be compatible with Debian stable's [2.36](https://packages.debian.org/source/stable/glibc).

These will also be the OSs for release builds btw.